readme.md

```diff
@@ -1,50 +1,79 @@
 
+# vdom-streaming-serializer
 
+The vdom-streaming-serializer is a library for serializing virtual DOM objects into chunks of strings that can be written into an HTTP response.
 
-## Usage
-
-### ES6 use
+## Installation
+```
+npm install vdom-streaming-serializer
+```
 
-With StealJS, you can import this module directly in a template that is autorendered:
+## Usage
+A virtual dom has to be created and running the serialization will emit streams of asynchronous elements.
 
+### Example
+This example creates a virtual dom and serializes it to emit streams of html elements in Chunks.
 ```js
-import plugin from 'vdom-streaming-serializer';
+var document = makeDocument();
+var ul = document.createElement('ul');
+document.body.appendChild(ul);
+var li = document.createElement('li');
+```
+Making this ```html <li> ``` as asynchronous will force the serialize to wait.
+```js
+var ASYNC = Symbol.for('async-node');
+li[ASYNC] = Promise.resolve();
+ul.appendChild(li);
 ```
 
-### CommonJS use
-
-Use `require` to load `vdom-streaming-serializer` and everything else
-needed to create a template that uses `vdom-streaming-serializer`:
-
+Once the element is serialized, it is then emitted in ```stream.on```.
 ```js
-var plugin = require("vdom-streaming-serializer");
+var stream = serialize(document);
+stream.setEncoding('utf8');
+stream.on('data', function(html){
+	console.log('Chunk', html);	
+});
+```
+The output from this demo should be:
 ```
+Chunk <html><body><ul>
+Chunk <li></li></ul<body></html>
+```
+From this, the ```html <li> ``` element is serialized asynchronously.
 
-## AMD use
 
-Configure the `can` and `jquery` paths and the `vdom-streaming-serializer` package:
+## API
+The serialization process traverses and checks for html syntax and attributes to add to the buffer.
+```js
+var generator = serialize(child);
+var result = generator.next();
+while(!result.done) {
+	buffer += result.value.buffer;
+	if(result.value.node && result.value.node[ASYNC]) {
+		yield {
+			buffer: buffer,
+			node: result.value.node
+		};
+		buffer = '';
+	}
+	result = generator.next();
+}
+child = child.nextSibling;
+```
+Upon noticing a child node, the serializer determines whether its child is asynchronous or not. If it is, the current buffer is canceled and the element begins serializing the asynchronous child.
 
-```html
-<script src="require.js"></script>
-<script>
-	require.config({
-	    paths: {
-	        "jquery": "node_modules/jquery/dist/jquery",
-	        "can": "node_modules/canjs/dist/amd/can"
-	    },
-	    packages: [{
-		    	name: 'vdom-streaming-serializer',
-		    	location: 'node_modules/vdom-streaming-serializer/dist/amd',
-		    	main: 'lib/vdom-streaming-serializer'
-	    }]
-	});
-	require(["main-amd"], function(){});
-</script>
+### Asynchronization
+It handles synchronous and asynchronous elements differently by resolving promise elements if we wish to wait before emitting.
+```js
```

Comment:
Would remove this snippet too, since it shows how the internal code works. But great explanation of what it does.